modules/Bio/EnsEMBL/Hive/HivePipeline.pm

```diff
@@ -544,151 +544,203 @@ sub print_diagram {
 sub apply_tweaks {
     my $self    = shift @_;
     my $tweaks  = shift @_;
-
+    my @responce;
+    my $responceStructure;
     my $need_write = 0;
-
+    $responceStructure->{Tweaks} = [];
     foreach my $tweak (@$tweaks) {
-        print "\nTweak.Request\t$tweak\n";
+        push @responce, "Tweak.Request  \t$tweak\n";
 
         if($tweak=~/^pipeline\.param\[(\w+)\](\?|#|=(.+))$/) {
             my ($param_name, $operator, $new_value_str) = ($1, $2, $3);
-
+            my $tweakStructure;
+            $tweakStructure->{Object}->{Type} = 'Pipeline';
+            $tweakStructure->{Object}->{Id} = '';
+            $tweakStructure->{Return}->{Field} = $param_name;
             my $pwp_collection  = $self->collection_of( 'PipelineWideParameters' );
             my $hash_pair       = $pwp_collection->find_one_by('param_name', $param_name);
-
             if($operator eq '?') {
-                print "Tweak.Show    \tpipeline.param[$param_name] ::\t"
-                     . ($hash_pair ? $hash_pair->{'param_value'} : '(missing_value)') . "\n";
+                my $value = $hash_pair ? $hash_pair->{'param_value'} : '(missing_value)';
+                $tweakStructure->{Action} = 'SHOW';
+                $tweakStructure->{Return}->{OldValue} = $value;
+                $tweakStructure->{Return}->{NewValue} = $value;
```

Comment:
For SHOW actions, `NewValue` is redundant with `OldValue`. Can we just use `Value` ?